distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso

```diff
@@ -142,3 +162,48 @@ resolve_aggregate table problem_builder aggregate_column =
 ## PRIVATE
    A marker for missing columns during resolution.
 type Internal_Missing_Column_Error
+
+## PRIVATE
+   Creates a Java AggregateColumnDefinition for the Aggregate_Column
+java_aggregator : Aggregate_Column->AggregateColumnDefinition
+java_aggregator name column =
+    case column of
+        Group_By c _ -> GroupByColumn.new name c.java_column
+        Count _ -> CountColumn.new name
+        Count_Distinct columns _ ignore_nothing ->
+            resolved = columns.map .java_column
+            CountDistinctColumn.new name resolved.to_array ignore_nothing
+        Count_Not_Nothing c _ -> CountNothingColumn.new name c.java_column False
+        Count_Nothing c _ -> CountNothingColumn.new name c.java_column True
+        Count_Not_Empty c _ -> CountEmptyColumn.new name c.java_column False
+        Count_Empty c _ -> CountEmptyColumn.new name c.java_column True
+        Sum c _ -> SumColumn.new name c.java_column
+        Average c _ -> MeanColumn.new name c.java_column
+        Standard_Deviation c _ p -> StandardDeviationColumn.new name c.java_column p
+        Median c _ -> PercentileColumn.new name c.java_column 0.5
+        Percentile p c _ -> PercentileColumn.new name c.java_column p
+        Mode c _ -> ModeColumn.new name c.java_column
+        First c _ ignore_nothing _ -> FirstColumn.new name c.java_column ignore_nothing
+        Last c _ ignore_nothing _ -> LastColumn.new name c.java_column ignore_nothing
+        Maximum c _ -> MinOrMaxColumn.new name c.java_column 1
+        Minimum c _ -> MinOrMaxColumn.new name c.java_column -1
+        Shortest c _ -> ShortestOrLongestColumn.new name c.java_column -1
+        Longest c _ -> ShortestOrLongestColumn.new name c.java_column 1
+        Concatenate c _ join prefix suffix quote -> ConcatenateColumn.new name c.java_column join prefix suffix quote
+        _ -> Error.throw "Unsupported aggregation."
+
+## PRIVATE
+   Convert Java aggregated problems to Enso Vector of equivalents
+parse_aggregated_problems : Any->Vector
+parse_aggregated_problems problems =
+    if problems.is_nothing then [] else
+        problems_array = problems.getProblems
+        parsed = Vector.new problems_array.length i->
+            p = problems_array.at i
+            if Java.is_instance p InvalidAggregation then Invalid_Aggregation p.getColumnName p.getRow p.getMessage else
+                if Java.is_instance p FloatingPointGrouping then Floating_Point_Grouping p.getColumnName p.getRow else
+                    if Java.is_instance p UnquotedDelimiter then Unquoted_Delimiter p.getColumnName p.getRow else
+                        "Unknown Error"
```

Comment:
```suggestion
                        Illegal_State_Error "Could not recognize a reported problem coming from the Java `org.enso.table` library."
```

I think we shouldn't report warnings as pure Text, much like with dataflow errors - it makes it hard to match specific types etc. If a generic one or an 'impossible state' I think `Illegal_State_Error` is a decent fallback. Also I'd add some detail so that we know what has happened - `Unknown error` is rather vague.